### DISCUSSION TITLE
Pull secrets for initImage and Broker image

### DISCUSSION THREAD

**Discussion by lordigon** [arkmq-org/arkmq-org-broker-operator#447] [Q&A] [answered]
Hi all I need to pull a custom initImage and a custom broker image from my own repository that needs credentials. Is it possible to specify a secret to define docker registry access credentials? Do you have any example to follow?

**Comment by brusdev** [comment, marked as answer]
The operator doesn't provide a way to define the `imagePullSecrets` field. To use a private registry you have the following options:

- Configuring Nodes to Authenticate to a Private Registry
  all pods can read any configured private registries
  requires node configuration by cluster administrator
- Pre-pulled Images
  all pods can use any images cached on a node
  requires root access to all nodes to set up

For further details see https://kubernetes.io/docs/concepts/containers/images/#using-a-private-registry

If those options doesn't work for you, feel to create a new issue, community contributions are always welcome :-)

**Comment by ivanavguston-oviva** [comment]
Hi, I think I have implemented the feature for `imagePullSecrets` How to contribute to your project or create PR?

**Reply by brusdev** [reply]
Hi @ivanavguston-oviva , yes the best way to contribute it to create an issue to describe your feature, i.e.  https://github.com/artemiscloud/activemq-artemis-operator/issues/562 and a PR related the new issue.

Feel free to tag me if you need any help, any contribute is welcome!!!

**Reply by brusdev** [reply]
@ivanavguston-oviva actually I see that a PR has been already created https://github.com/artemiscloud/activemq-artemis-operator/pull/560

**Reply by brusdev** [reply]
PR #560 is replaced by #582
